### ISSUE TITLE
golangci-lint v1.24.0 still appears to not be cleaning up lock file 

### ISSUE THREAD

**Issue by SteelPhase** [golangci/golangci-lint#1021]
Lock file should be cleaned up when running `golangci-lint --version` or `golangci-lint --help`

<details><summary>Version of golangci-lint v1.24.0</summary>

```console
$ golangci-lint --version
golangci-lint has version 1.24.0 built from 6fd4383 on 2020-03-15T11:38:02Z
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.14.1 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/steelphase/.cache/go-build"
GOENV="/home/steelphase/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOOS="linux"
GOPATH="/home/steelphase/go"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build950404935=/tmp/go-build -gno-record-gcc-switches"
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ ls -la /tmp/golangci-lint.lock
ls: cannot access /tmp/golangci-lint.lock: No such file or directory
$ golangci-lint -v --version
INFO [config_reader] Config search paths: [./ /home/steelphase /home /] 
golangci-lint has version 1.24.0 built from 6fd4383 on 2020-03-15T11:38:02Z
$ ls -la /tmp/golangci-lint.lock
-rw-------. 1 steelphase users 0 Apr 15 15:46 /tmp/golangci-lint.lock
```

</details>

**Comment by jirfag** [comment]
Hi,
thank you,
can't reproduce. Please, re-run `golangci-lint` with environment variable `GL_DEBUG=exec`

**Comment by SteelPhase** [comment]
Just tested on linux, and got the below output. looks like `golangci-lint version` is fine, but `gloangci-lint --version` is not

```
$ golangci-lint version
golangci-lint has version 1.26.0 built from 6bd10d0 on 2020-05-01T15:33:57Z
$ export GL_DEBUG=exec
$ ls -la /tmp/golangci-lint.lock
ls: cannot access /tmp/golangci-lint.lock: No such file or directory
$ golangci-lint -v --version
DEBU [exec] Starting execution...                 
DEBU [exec] Locking on file /tmp/golangci-lint.lock... 
INFO [config_reader] Config search paths: [./ /home/steelphase /home /] 
DEBU [exec] Initialized executor                  
golangci-lint has version 1.26.0 built from 6bd10d0 on 2020-05-01T15:33:57Z
$ ls -la /tmp/golangci-lint.lock
-rw-------. 1 steelphase users 0 May  5 14:55 /tmp/golangci-lint.lock
$ golangci-lint -v version
DEBU [exec] Starting execution...                 
DEBU [exec] Locking on file /tmp/golangci-lint.lock... 
INFO [config_reader] Config search paths: [./ /home/steelphase /home /] 
DEBU [exec] Initialized executor                  
golangci-lint has version 1.26.0 built from 6bd10d0 on 2020-05-01T15:33:57Z
$ ls -la /tmp/golangci-lint.lock
ls: cannot access /tmp/golangci-lint.lock: No such file or directory
```

__NOTE:__ I had to go get because the binary in golangci-lint-1.26.0-freebsd-386.tar.gz refused to work on my machine

On mac:
```$GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.26.0
go: found github.com/golangci/golangci-lint/cmd/golangci-lint in github.com/golangci/golangci-lint v1.26.0
$export GL_DEBUG=exec
$ls -la  /var/folders/0b/00/T/golangci-lint.lock
ls: /var/folders/0b/00/T/golangci-lint.lock: No such file or directory
$golangci-lint -v --version
DEBU [exec] Starting execution...                 
DEBU [exec] Locking on file /var/folders/0b/00/T/golangci-lint.lock... 
INFO [config_reader] Config search paths: [./ /Users/steelphase /Users /] 
DEBU [exec] Initialized executor                  
golangci-lint has version v1.26.0 built from (unknown, mod sum: "h1:CLLGRSA9BLMiNvsWPXHioYAdfIx9tkgdVWyA6bIdYCo=") on (unknown)
$ls -la  /var/folders/0b/00/T/golangci-lint.lock
-rw-------  1 steelphase  group  0 May  5 15:03 /var/folders/0b/00/T/golangci-lint.lock
$golangci-lint -v version
DEBU [exec] Starting execution...                 
DEBU [exec] Locking on file /var/folders/0b/00/T/golangci-lint.lock... 
INFO [config_reader] Config search paths: [./ /Users/steelphase /Users /] 
DEBU [exec] Initialized executor                  
golangci-lint has version v1.26.0 built from (unknown, mod sum: "h1:CLLGRSA9BLMiNvsWPXHioYAdfIx9tkgdVWyA6bIdYCo=") on (unknown)
$ls -la  /var/folders/0b/00/T/golangci-lint.lock
ls: /var/folders/0b/00/T/golangci-lint.lock: No such file or directory
```

**Comment by SteelPhase** [comment]
I would say this is the issue for --version

https://github.com/golangci/golangci-lint/blob/6bd10d01fde78697441d9c11e2235f0dbb1e2822/pkg/commands/root.go#L19-L22

**Comment by ksoichiro** [comment]
I confirmed that it can be reproduced on v1.26.0, but not on v1.27.0.
It seems it's already fixed by #1070, which is included in the latest released version: v1.27.0.
So maybe you can resolve this by updating to v1.27.0.

As @SteelPhase pointed out, there were some paths that exits without removing lock file. Removing lock file  is executed by cobra.Command's PostRun hook, but it is not set for those cases.
After #1070, it seems those paths don't exist because it is changed to acquiring file lock on PreRun and releasing it on PostRun.

**Comment by SVilgelm** [comment]
closing due to the issue is fixed in the v1.27.0 release
